src/services/api/api.service.ts

```diff
@@ -90,16 +116,56 @@ async function attachHeaders(ctx: Context<any, any>, res: ServerResponse) {
   res.setHeader("X-Query-At", new Date().toISOString());
 }
 
+function createOnBeforeCall(required: boolean = true) {
+  return async function (
+    ctx: Context<any, any>,
+    _route: Route,
+    req: IncomingMessage
+  ) {
+    await parseAtBlockHeight(ctx, req, required);
+  };
+}
 
-const ensureBlockHeightForIndexer = async function (
-  ctx: Context<any, any>,
-  _route: Route,
-  req: IncomingMessage
+function createOnError() {
+  return function (
+    req: IncomingMessage,
+    res: ServerResponse,
+    err: any
+  ) {
+    const status = err.code || 428;
```

Comment:
Although it is not incorrect, I think 400 is a more typical status code for this situation.